### PR TITLE
Support fluentd v0.14

### LINKED ISSUE #5: Could you migrate to Fluentd v0.14 API?
Could you migrate to Fluent d v0.14 API?

* Pros/Cons
  * https://www.clear-code.com/blog/2017/5/23.html (ja)
  * https://gist.github.com/okkez/05ee6af7b8bb7de852a36eb2f679c344 (en)
* Updating plugin for v0.14 from v0.12
  * http://docs.fluentd.org/v0.14/articles/plugin-update-from-v12

Other chat(like) services:
* fluent-plugin-hipchat: done
* fluent-plugin-slack: will be merged soon
* fluent-plugin-twitter: done

And 100+ plugins have already been migrated to Fluentd v0.14 plugin API.

Mention me, please. If you have questions.

### PR DESCRIPTION
**Pull request by tsuyoshizawa**
fix #5

### CODE REVIEW

File: test/plugin/test_out_typetalk.rb [hunk 1]
```diff
@@ -70,21 +72,23 @@ def test_configure_no_throttle
 
   def test_write
     d = create_driver()
-    mock(d.instance.typetalk).post_message(1, 'notice : test1')
-    d.emit({'message' => 'test1'})
-    d.run()
+    stub(d.instance.typetalk).post_message(1, 'notice : test1')
```
Comment (okkez) [inline test/plugin/test_out_typetalk.rb]: `stub` does not assert method invocation. I think `mock` is prefer here.

File: test/plugin/test_out_typetalk.rb [hunk 2]
```diff
@@ -70,21 +72,23 @@ def test_configure_no_throttle
...
+    end
   end
 
   def test_template
-    d = create_driver(CONFIG, 'warn')
+    d = create_driver(CONFIG)
     d.instance.message = "notice : %s [%s]"
     d.instance.out_keys = ["message", "time"]
-    mock(d.instance.typetalk).post_message(1, "notice : test1 [1399910738]")
+    stub(d.instance.typetalk).post_message(1, "notice : test1 [1399910738]")
```
Comment (okkez) [inline test/plugin/test_out_typetalk.rb]: ditto.

File: test/helper.rb [hunk 3]
```diff
@@ -1,29 +1,8 @@
...
-rescue Bundler::BundlerError => e
-  $stderr.puts e.message
-  $stderr.puts "Run `bundle install` to install missing gems"
-  exit e.status_code
-end
-require 'test/unit'
-require 'test/unit/rr'
-
 $LOAD_PATH.unshift(File.join(File.dirname(__FILE__), '..', 'lib'))
 $LOAD_PATH.unshift(File.dirname(__FILE__))
```
Comment (okkez) [inline test/helper.rb]: How about `__dir__` instead of `File.dirname(__FILE__)`?

File: test/helper.rb [hunk 4]
```diff
@@ -1,29 +1,8 @@
-require 'rubygems'
-require 'bundler'
-begin
-  Bundler.setup(:default, :development)
-rescue Bundler::BundlerError => e
-  $stderr.puts e.message
-  $stderr.puts "Run `bundle install` to install missing gems"
-  exit e.status_code
-end
```
Comment (okkez) [inline test/helper.rb]: Is `require "bundler/setup"` missing?

File: fluent-plugin-typetalk.gemspec [hunk 5]
```diff
@@ -18,14 +17,11 @@ Gem::Specification.new do |spec|
   spec.require_paths = ["lib"]
 
   spec.add_development_dependency "bundler", "~> 1.3"
-  spec.add_development_dependency "rake"
+  spec.add_development_dependency "rake", ">= 0.9.2"
   spec.add_development_dependency "rr", ">= 1.0.0"
```
Comment (okkez) [inline fluent-plugin-typetalk.gemspec]: We can remove this line. Because we depends on `test-unit-rr`.

**Comment by tsuyoshizawa** [comment]
@okkez Thank you for review.

I fixed codes, and I will squash my commits if your feedback is now clear.

File: README.md [hunk 6]
```diff
@@ -4,6 +4,13 @@
 
 [Fluentd](http://fluentd.org) plugin to emit notifications to [Typetalk](http://typetalk.in/).
 
+## Requirements
+
+| fluent-plugin-hipchat | fluentd | ruby |
```
Comment (okkez) [inline README.md]: fluent-plugin-typetalk?

File: test/plugin/test_out_typetalk.rb [hunk 7]
```diff
@@ -1,10 +1,12 @@
 # coding: utf-8
 
 require 'helper'
+require 'fluent/plugin/out_typetalk'
 
 class TypetalkOutputTest < Test::Unit::TestCase
-  
+
   def setup
+    super
```
Comment (okkez) [inline test/plugin/test_out_typetalk.rb]: I think we can remove `super`.

**Comment by okkez** [comment]
LGTM :100:

File: README.md (hunk 6 above)
Comment (tsuyoshizawa) [inline README.md]: Oops... Thank you!
I referred hipchat plugin to support fluentd v0.14.

**Pull request merged**